ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/searchtree/algorithms/SearchTree.java

```diff
@@ -99,11 +100,15 @@ public SearchTree(SearchTreeInput input,
     public CompletableFuture<OptimizationResult> run() {
         String preSearchTreeVariantId = input.getNetwork().getVariantManager().getWorkingVariantId();
         input.getNetwork().getVariantManager().cloneVariant(preSearchTreeVariantId, SEARCH_TREE_WORKING_VARIANT_ID, true);
-        input.getNetwork().getVariantManager().setWorkingVariant(SEARCH_TREE_WORKING_VARIANT_ID);
+        input.getNetwork().getVariantManager().setWorkingVariant(SEARCH_TREE_WORKING_VARIANT_ID); // the variant used for root leaf and all the child leaves
         try {
             initLeaves(input);
 
             TECHNICAL_LOGS.debug("Evaluating root leaf");
+
```

Comment:
is it possible to only run loadflow when necessary, by tracking if there are hvdc network actions defined for instance ? 
also, add a debug log to inform we're running a loadflow